Quick edit game player character

nazar opened the pull request:
References #163

nazar committed:
closes #163 - quick edit game player character

nazar commented:
I'm going to check out React Semantic docs to see if it supports the new 16.3 Context in the Popup component as I'm not a fan in how I'm passing the `gameId` down the the modal.

nazar commented:
It doesn't look like [the code for React 16.3](https://github.com/Semantic-Org/Semantic-UI-React/pull/2880) has been released yet.

I'll keep an 👁 on semantic and update once React 16.3 changes are released.

nazar committed:
#163 - tidy the quick edit form

mikelax commented on api/src/services/characters/quickUpdateCharacterAndAddUpdateGameMessage.js:
```diff
@@ -0,0 +1,91 @@
+import serviceExecutor from 'utils/serviceExecutor';
+
+
+export default function ({
```
@nazar  should this be a named function? is there an eslint rule for this?

nazar replied:
A function name is optional when it is exported as the default function.

mikelax reviewed (commented):
left one small comment. 
Also, where is the logic that looks at the type of game to decide what fields should be included in the quick edit modal?

nazar committed:
#163 - various

1. Add per label quick edit forms
2. split up updateCharacter and quickUpdateCharacter mutations to avoid generalization rabbit hole

mikelax commented on api/src/services/characters/quickUpdateCharacterAndAddUpdateGameMessage.js:
```diff
@@ -0,0 +1,91 @@
+      .chain()
+      .map((change) => {
+        return (`
+          <div class="change-meta">
```
@nazar  should this be a `p` instead of `div` to match the messages generated by the editor?
Also, when I look at the html that was saved to the game_messages table, I don't see the class attribute, is it possible that is getting stripped somewhere?

mikelax commented on api/src/services/characters/quickUpdateCharacterAndAddUpdateGameMessage.js:
```diff
@@ -0,0 +1,91 @@
+      .map((change) => {
+        return (`
+          <div class="change-meta">
+            <strong>${changeDescription || 'Changed'}</strong>: 
```
@nazar  should the value here `changeDescription` be stored in the character_logs table in the change_description field? I think it should, right? 

Eventually we are going to allow viewing the "history", or changelog of a Character, and it would be nice to see the the human entered value for the description rather than the generic `Character stats updated`.

nazar replied:
>should the value here changeDescription be stored in the character_logs table in the change_description field? I think it should, right?

`changeDescription` should be stored `character_logs` - I'll double check 👍

nazar replied:
Nice catch @mikelax

mikelax commented on api/src/services/characters/quickUpdateCharacterAndAddUpdateGameMessage.js:
```diff
@@ -0,0 +1,91 @@
+  function doUpdateCharacter() {
+    const cleanInput = _(input)
+      .chain()
+      .omit(['changeDescription', 'changeMeta'])
```
@nazar should `changeDescription` be removed from this line here so the user entered description will override the default `Character stats updated` message?

Update:  I tried removing the field but then the save function doesn't work, so I must be missing the reason for it.

nazar committed:
#163 - various

1. fix saving of change description in characters_logs
2. only character owners can see the quick exit popup
3. fix owner guard not working on characters edit

nazar committed:
#163 - introduce an icm post type to describe in character system Messages. This extra type is used to hide the "edit" button on character update game messages

mikelax commented on api/package.json:
```diff
@@ -50,6 +50,7 @@
     "source-map-support": "0.5.4",
     "sql-fixtures": "1.0.0",
     "subscriptions-transport-ws": "0.9.7",
+    "tsml": "^1.0.1",
```
@nazar  I'll merge this PR, but do you think this lib is necessary? I saw its not maintained anymore, though I assume the function it performs is really straight forward so no need for updates??

nazar replied:
The lib is nice in that it strips whitespace and newlines added in backtick quoted multiline strings.

The [code is < 10 lines](https://github.com/rvagg/archived-tsml/blob/master/tsml.js) - I did attempt to copypasta it but eslint disliked the non fancy es6 formats and I was too short on time to es5 -> es6 it.
